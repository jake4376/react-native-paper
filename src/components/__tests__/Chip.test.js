/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import Chip from '../Chip';

it('renders chip with onPress', () => {
  const tree = renderer
    .create(<Chip onPress={() => {}}>Example Chip</Chip>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders chip with icon', () => {
  const tree = renderer.create(<Chip icon="info">Example Chip</Chip>).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders deletable chip', () => {
  const tree = renderer
    .create(
      <Chip icon="info" onDelete={() => {}}>
        Example Chip
      </Chip>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
