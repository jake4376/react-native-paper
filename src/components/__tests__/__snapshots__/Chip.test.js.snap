// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders chip with icon 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 16,
        "flexDirection": "row",
        "justifyContent": "space-around",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    pointerEvents="none"
    style={
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 32,
        },
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "marginVertical": 8,
          },
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "marginLeft": 8,
            "marginRight": 12,
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    Example Chip
  </Text>
</View>
`;

exports[`renders chip with onPress 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 16,
        "flexDirection": "row",
        "justifyContent": "space-around",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "marginVertical": 8,
          },
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "marginLeft": 12,
            "marginRight": 12,
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    Example Chip
  </Text>
</View>
`;

exports[`renders deletable chip 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 16,
        "flexDirection": "row",
        "justifyContent": "space-around",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.12)",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    pointerEvents="none"
    style={
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 32,
        },
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "marginVertical": 8,
          },
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "marginLeft": 8,
            "marginRight": 0,
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#03dac4",
          "background": "#f6f6f6",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "error": "#B00020",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#6200ee",
          "surface": "#ffffff",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 4,
      }
    }
  >
    Example Chip
  </Text>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "padding": 6,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      pointerEvents="none"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 20,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
